riscv/processor.cc

```diff
@@ -949,7 +950,8 @@ reg_t processor_t::get_csr(int which, insn_t insn, bool write, bool peek)
 
 reg_t illegal_instruction(processor_t* p, insn_t insn, reg_t pc)
 {
-  throw trap_illegal_instruction(insn.bits());
+  // the illegal instruction may longer than ILEN bits
+  throw trap_illegal_instruction(insn.bits() & ((reg_t(1)<<(p->get_ilen()))-1));
```

Comment:
This line is the only functional change here, yes? The rest looks like a lot of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) only to ultimately hard-code an ilen of 32.

Why not just hard-code the 32-bit ilen here, and leave a comment to that effect.

Reply:
Unrelated comment: this will overflow the shift amount for ILEN=64. Can be avoided by writing (2<<(x-1))-1 instead of (1<<x)-1. But probably irrelevant for the reason Scott mentioned.

Reply:
@aswaterman @scottj97 
Thanks for your comments. Since there is no instruction longer than 32 bits, I think simplicity assuming 32-bit ILEN makes sense for most implementations.
I updated the patch and commit log with your comments, which avoids complex modification by hard-coding the ILEN.